spec/requests/alert_definitions_spec.rb

```diff
@@ -156,6 +216,77 @@
     expect(response.parsed_body).to include(expected)
   end
 
+  it "edits an alert definition with miq_expression" do
+    api_basic_authorize(action_identifier(:alert_definitions, :edit, :resource_actions, :post))
+    alert_definition = FactoryGirl.create(
+      :miq_alert,
+      :miq_expression => MiqExpression.new(:eval_method => "mw_heap_used", :mode => "internal", :options => {}),
+      :options        => { :notifications => {:delay_next_evaluation => 0, :evm_event => {} } }
+    )
+
+    exp = { :eval_method => "nothing", :mode => "internal", :options => {} }
+
+    post(
+      api_alert_definition_url(nil, alert_definition),
+      :params => {
+        :action          => "edit",
+        :miq_expression  => nil,
+        :hash_expression => exp
+      }
+    )
+
+    expect(response).to have_http_status(:ok)
+    expect(response.parsed_body).to include("hash_expression" => exp.stringify_keys)
+  end
+
+  it "edits an alert definition with hash_expression" do
+    api_basic_authorize(action_identifier(:alert_definitions, :edit, :resource_actions, :post))
+    alert_definition = FactoryGirl.create(
+      :miq_alert,
+      :hash_expression => { :eval_method => "nothing", :mode => "internal", :options => {} },
+      :options         => { :notifications => {:delay_next_evaluation => 0, :evm_event => {} } }
+    )
+
+    exp = { :eval_method => "mw_heap_used", :mode => "internal", :options => {} }
+
+    post(
+      api_alert_definition_url(nil, alert_definition),
+      :params => {
+        :action          => "edit",
+        :miq_expression  => exp,
+        :hash_expression => nil
+      }
+    )
+
+    expect(response).to have_http_status(:ok)
+    expect(response.parsed_body).to include(
+      "miq_expression" => {
+        "exp"          => exp.stringify_keys,
+        "context_type" => nil
+      }
+    )
+  end
+
+  it "fails to edit an alert definition with more than one expression" do
+    api_basic_authorize(action_identifier(:alert_definitions, :edit, :resource_actions, :post))
+    alert_definition = FactoryGirl.create(
+      :miq_alert,
+      :hash_expression => { :eval_method => "nothing", :mode => "internal", :options => {} },
+      :options         => { :notifications => {:delay_next_evaluation => 0, :evm_event => {} } }
+    )
+
+    post(
+      api_alert_definition_url(nil, alert_definition),
+      :params => {
+        :action          => "edit",
+        :hash_expression => { :eval_method => "event_threshold", :mode => "internal", :options => {} },
+        :miq_expression  => { :eval_method => "mw_heap_used", :mode => "internal", :options => {} }
```

Comment:
As above

Reply:
It looks like this is still a "hash_expression" in place of an "miq_expression"?

Reply:
oh, you're right, I didn't actually fix this one when I said it was. I fixed it now.